src/jit/hwintrinsicxarch.cpp

```diff
@@ -755,6 +761,89 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
                     gtNewSimdHWIntrinsicNode(TYP_SIMD16, op2, op1, NI_SSE2_CompareGreaterThan, baseType, simdSize);
             }
             break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Double:
+        {
+            assert(sig->numArgs == 2);
+            assert(getBaseTypeOfSIMDType(sig->retTypeSigClass) == TYP_DOUBLE);
+
+            argList = info.compCompHnd->getArgNext(sig->args);
+            CorInfoType corType =
+                strip(info.compCompHnd->getArgType(sig, argList, &argClass)); // type of the second argument
+
+            baseType = JITtype2varType(corType);
+
+#ifdef _TARGET_X86_
+            if (varTypeIsLong(JITtype2varType(corType)))
+            {
+                return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
+            }
+#endif // _TARGET_X86_
+
+            if (baseType == TYP_STRUCT)
```

Comment:
I think we could simplify this to use `getArgForHWIntrinsic`

Reply:
Will do

Reply:
Unfortunately, `getArgForHWIntrinsic` does not handle this logic properly

Reply:
What about it isn't being handled properly?

Reply:
It returns `TYP_STRUCT` instead of `TYP_FLOAT` - didn't dig deeper to solve it. Plan to have closer look while working on #16329